### PR TITLE
Accessibility updates to codelab.md, accessibility.md

### PR DESCRIPTION
**Pull request by pandeymauli**
_Description of what this PR is changing or adding, and why:_

- Updates to codelab.md to show how the startup-namer app (part 1) can be made accessible
- Adds a new subsection titled 'Testing accessibility on mobile' to accessibility.md to demonstrate the use of AccessibilityGuideline API. The section includes a code sample to show how to write unit tests using the API.
- Removed semi-colon from 'Testing accessibility on web:' on accessibility.md

_Issues fixed by this PR (if any):_ Fixes #ISSUE-NUMBER

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/get-started/codelab.md [hunk 1]
```diff
@@ -743,6 +812,12 @@ where you add the following functionality:
   containing the saved favorites.
 * Modify the theme color, making an all-white app.
 
+## Explore the Flutter SDK
+
+* <a href="https://docs.flutter.dev/get-started/flutter-for/react-native-devs">Flutter for React Native developers</a>
```
Comment (domesticmouse) [inline src/get-started/codelab.md:817]: Was this change intended? Seems unrelated to the rest of the change.

**Review by domesticmouse — COMMENTED** [review]
I'm guessing this PR is waiting for the codelab changes to land and be included?

**Comment by domesticmouse** [comment]
PTAL @khanhnwin

File: src/development/accessibility-and-localization/accessibility.md [hunk 2]
```diff
@@ -144,7 +144,34 @@ widgets to offer a dramatically more accessible experience.
...
 
-## Testing accessibility on web:
+## Testing accessibility on mobile
+
+You can use Flutter's <a href="https://api.flutter.dev/flutter/flutter_test/AccessibilityGuideline-class.html">Accessibility Guideline API</a> to test if your mobile app's UI meets the accessibility recommendations for text contrast, target size, and target labels. 
+
+The example below shows how to use the Guideline API on <a href="https://docs.flutter.dev/get-started/codelab/?tab=talkback">Startup Name Generator</a>, the app built as part of the codelab on creating your first Flutter app. Each list tile on the app's main screen is tappable target with text represented in 18 point. You can add the tests provided by the Guideline API in `test/widget_test.dart` of your app directory.
+
+<?code-excerpt path-base="codelabs/startup_namer/step3_stateful_widget"?>
+<?code-excerpt "test/widget_test.dart" title indent-by="2"?>
```
Comment (khanhnwin) [inline src/development/accessibility-and-localization/accessibility.md:154]: It looks like this code excerpt is out of date. 

When I run [`./tool/refresh-code-excerpts.sh`](examples/codelabs/startup_namer/step3_stateful_widget/test/widget_test.dart), it replaces the entire snippet with the contents from the file at : `examples/codelabs/startup_namer/step3_stateful_widget/test/widget_test.dart`

Mauli, you'll have to integrate that excerpt into that widget_test.dart file in the codelabs repo (cc @domesticmouse), land it, and then pull in the updated codelab submodule into this PR. The refresh code excerpts should be happy when that's completed.

Comment (domesticmouse) [reply src/development/accessibility-and-localization/accessibility.md:154]: The updated codelab that should land in the next update cycle will contain @pandeymauli's updated `widget_test.dart`, so hopefully everything just works?

**Comment by khanhnwin** [comment]
Hey @pandeymauli! I just refreshed the code excerpts - you'll want to add the code excerpt start/end markers that we talked about for the relevant code section in the widget_test.dart file! Right now it's pulling in the entire file.

**Comment by pandeymauli** [comment]
Thanks, Khanh! I just added those markers to the code excerpt and created
another pull request.

**Comment by khanhnwin** [comment]
Thanks for pushing this over the finish line @pandeymauli @domesticmouse! I just added the excerpt title and that should be the last of it. We can land once the build passes.

Mauli, I'll follow up with you offline once I hear back about your video upload!

File: src/get-started/codelab.md (hunk 1 above)
Comment (khanhnwin) [reply src/get-started/codelab.md:817]: Build is green! We should be good to land once this comment is resolved.

Comment (domesticmouse) [reply src/get-started/codelab.md:817]: I'd revert this part of the change as it's an unrelated change. If it's required, please fold into a new PR. Alternatively, it might be guff from Git history shenanigans...

**Comment by sfshaza2** [comment]
I really appreciate @domesticmouse and @Khanh Nguyen
***@***.***> helping
you with this PR, @Mauli Pandey ***@***.***>. But, for the
record, the FIRST thing you should do in this situation, is merge your
changes with the latest version of the repo. Our website is far from static.

File: src/development/accessibility-and-localization/accessibility.md [hunk 3]
```diff
@@ -144,7 +144,34 @@ widgets to offer a dramatically more accessible experience.
 
 <iframe width="560" height="315" src="{{site.youtube-site}}/embed/bWbBgbmAdQs" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 
-## Testing accessibility on web:
+## Testing accessibility on mobile
+
+You can use Flutter's <a href="https://api.flutter.dev/flutter/flutter_test/AccessibilityGuideline-class.html">Accessibility Guideline API</a> to test if your mobile app's UI meets the accessibility recommendations for text contrast, target size, and target labels. 
+
+The example below shows how to use the Guideline API on <a href="https://docs.flutter.dev/get-started/codelab/?tab=talkback">Startup Name Generator</a>, the app built as part of the codelab on creating your first Flutter app. Each list tile on the app's main screen is tappable target with text represented in 18 point. You can add the tests provided by the Guideline API in `test/widget_test.dart` of your app directory.
```
Comment (atsansone) [inline src/development/accessibility-and-localization/accessibility.md:151]: **issues:** 
- Replace "below" as it runs counter to [Google's documentation accessibility guidelines](https://developers.google.com/style/accessibility).
- Use [semantic line breaks](https://sembr.org/) for copy.

```suggestion
The following example shows how to use the Guideline API on
<a href="https://docs.flutter.dev/get-started/codelab/?tab=talkback">Startup Name Generator</a>,
the app built as part of the codelab on creating your first Flutter app.
Each list tile on the app's main screen serves as a tappable target
with text represented in 18 point. You can add Guideline API tests
in `test/widget_test.dart` of your app directory.
```

File: src/development/accessibility-and-localization/accessibility.md [hunk 4]
```diff
@@ -144,7 +144,34 @@ widgets to offer a dramatically more accessible experience.
 
 <iframe width="560" height="315" src="{{site.youtube-site}}/embed/bWbBgbmAdQs" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 
-## Testing accessibility on web:
+## Testing accessibility on mobile
+
+You can use Flutter's <a href="https://api.flutter.dev/flutter/flutter_test/AccessibilityGuideline-class.html">Accessibility Guideline API</a> to test if your mobile app's UI meets the accessibility recommendations for text contrast, target size, and target labels.
```
Comment (atsansone) [inline src/development/accessibility-and-localization/accessibility.md:149]: **issue:** Update copy with semantic line breaks and more concise copy.
```suggestion
Test your app using Flutter's
<a href="https://api.flutter.dev/flutter/flutter_test/AccessibilityGuideline-class.html">Accessibility Guideline API</a>.
This API checks if your app's UI meets Flutter's accessibility recommendations.
These cover recommendations for text contrast, target size, and target labels.
```

File: src/get-started/codelab.md [hunk 5]
```diff
@@ -216,6 +218,12 @@ where the Dart code lives.
   a `Text` child widget. The Center widget aligns its widget subtree
   to the center of the screen.
 
+{{site.alert.note}}
+      The app automatically creates a semantics tree for screen readers. 
+      Each node in the tree can correspond to one or several 
+      widgets and can be <a href="https://api.flutter.dev/flutter/widgets/Semantics-class.html">further customized</a> to tell screen readers how to behave with the node.
```
Comment (atsansone) [inline src/get-started/codelab.md:224]: **suggestion:** Make the copy more concise and use semantic line breaks.
```suggestion
      The app creates a semantics tree for screen readers. 
      Each node in the tree can correspond to one or several 
      widgets. These nodes can be <a href="https://api.flutter.dev/flutter/widgets/Semantics-class.html">customized</a>
      to tell screen readers how to behave with the node. 
```

File: src/get-started/codelab.md [hunk 6]
```diff
@@ -345,6 +353,12 @@ Process finished with exit code 0
       "UpperCamelCase".
     {{site.alert.end}}
 
+    {{site.alert.important}}
+    Using pascal case also helps screen readers identify the individual words 
+    in the compound word and provides a better 
+    experience to visually impaired users.
+    {{site.alert.end}}
```
Comment (atsansone) [inline src/get-started/codelab.md:360]: **issue:** Refer to this capitalization format as **Upper Camel Case** per the [Dart style guide](https://dart.dev/guides/language/effective-dart/style#do-name-types-using-uppercamelcase). Remove the mention of "pascal case".

File: src/get-started/codelab.md [hunk 7]
```diff
@@ -357,6 +371,54 @@ Process finished with exit code 0
       {% include docs/ios-windows-figure-pair.md image="step2.png" alt="App at completion of second step" %}
     {% endindent %}
 
+    **You can also try out the app with a screen reader. All you need to do is turn on the screen reader on your device by following the steps below:**
```
Comment (atsansone) [inline src/get-started/codelab.md:374]: **issues:** 
- Replace "below" as it runs counter to [Google's documentation accessibility guidelines](https://developers.google.com/style/accessibility).
- Use [semantic line breaks](https://sembr.org/) for copy.

```suggestion
    **You can try the app with a screen reader.
    Turn on the screen reader on your device using the following steps:** 
```

File: src/get-started/codelab.md [hunk 8]
```diff
@@ -357,6 +371,54 @@ Process finished with exit code 0
...
+    <div class="tab-pane active" id="talkback" role="tabpanel" aria-labelledby="talkback-tab" markdown="1">
+
+    <!-- **Turn on TalkBack on your Android phone to try the app with a screen reader** -->
+
+    1. On your device, open Settings. 
+    2. Select Accessibility and then TalkBack.
+    3. Turn 'Use TalkBack' on or off.
+    4. Select Ok.
+
+    You can also view this video to learn how to find and customize all accessibility features provided by Android.
```
Comment (atsansone) [inline src/get-started/codelab.md:398]: **issue:** [Avoid passive voice](https://developers.google.com/style/voice).

```suggestion
    To learn how to find and customize all accessibility features that Android provides, view this video.
```

**Review by atsansone — CHANGES_REQUESTED** [review]
@pandeymauli : Some style issues were found. Can you please revise and make a new commit?

**Comment by pandeymauli** [comment]
Just updated all the styling issues. My branch is up to date with the main. I can create a pull request but it may be better to wait for the video link. What do you think @khanhnwin and @domesticmouse?

**Comment by domesticmouse** [comment]
> Just updated all the styling issues. My branch is up to date with the main. I can create a pull request but it may be better to wait for the video link. What do you think @khanhnwin and @domesticmouse?

There are things in this PR that I'm uncomfortable with. The flutter and codelabs sub repos are showing as changed, and there is content that I believe should be deleted. I'm wondering if starting with a fresh PR and just putting in the changes you want without touching sub repos would wind up with a cleaner PR.

**Comment by domesticmouse** [comment]
Oh and: ERROR: some code excerpts were not refreshed!

**Comment by pandeymauli** [comment]
@domesticmouse I will wait for the video link before I submit a fresh PR.

File: src/get-started/codelab.md [hunk 9]
```diff
@@ -78,6 +79,7 @@ The animated GIF shows how the app works at the completion of part 1.
 
   * A physical device ([Android][] or [iOS][]) connected to your
     computer and set to developer mode
+  * Screen reader enabled on the physical device (<a href="#talkback">TalkBack</a> on Android, <a href="#talkback">VoiceOver</a> on iPhone)
   * The [iOS simulator][] (requires installing Xcode tools)
   * The [Android emulator][] (requires setup in Android Studio)
   * A browser (Chrome is required for debugging)
```
Comment (atsansone) [inline src/get-started/codelab.md:85]: **suggestion:** Active voice.
```suggestion
  * A browser (Debugging requires the Chrome browser.)
```

**Comment by khanhnwin** [comment]
Hey Mauli,

I just followed up with on the video upload. A few open questions on my end, once that's resolved I can send you the link!
